### ISSUE TITLE
reRNN Attention Instability / NaN

### ISSUE THREAD

**Issue by jason-fries** [HazyResearch/snorkel#623]
When specifying an attention window of any size > 0, I often encounter numerical problems that cause NaN or exploded values for the loss function. Training without attention enabled (`attn_window=0`) works fine. 

```
[reRNN] Epoch 26 (129.76s)	Average loss=0.263585
[reRNN] Epoch 27 (134.46s)	Average loss=0.466983
[reRNN] Epoch 28 (139.46s)	Average loss=1.090147
[reRNN] Epoch 29 (144.21s)	Average loss=87.524101
[reRNN] Epoch 30 (148.79s)	Average loss=1030.320557
[reRNN] Epoch 31 (153.32s)	Average loss=2438.344971
[reRNN] Epoch 32 (157.86s)	Average loss=23401.203125
[reRNN] Epoch 33 (162.37s)	Average loss=94097.406250
```

**Comment by jason-fries** [comment]
deprecated with move to pytorch.
